packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -237,7 +239,47 @@ export const Product = ({
       productId
     );
 
+    const webSubscriptions = (customerSubscriptions || []).filter((s) =>
+      isWebSubscription(s)
+    ) as WebSubscription[];
+
+    const alreadySubscribedToSelectedPlan = customerIsSubscribedToPlan(
+      webSubscriptions,
+      selectedPlan
+    );
+
+    const planUpdateEligibilityResult = subscriptionUpdateEligibilityResult(
+      webSubscriptions,
+      selectedPlan,
+      plansById,
+      config.featureFlags.useFirestoreProductConfigs
+    );
+
+    // iap subscription exists
     if (iapSubscription) {
+      // Returns appropriate iap roadblock error messaging
+      // if product/bundle is eligible for an upgrade, return contact support messaging
+      // since upgrade from web for an iap subscription is not currently supported
+      // else, return default (INVALID / BLOCKED_IAP) already subscribed through iap messaging
+      // n.b.[1]: BLOCKED_IAP is included in case the default changes and as an added check
+      // n.b.[2]: DOWNGRADE, and error code 'iap_upgrade_no_bundle_support' not yet supported
+      const iapErrorMessageCode = () => {
+        const iapSubscriptionUpdateEligibility =
+          getSubscriptionUpdateEligibility(
+            currentPlan,
+            selectedPlan,
+            config.featureFlags.useFirestoreProductConfigs
+          );
+
+        switch (iapSubscriptionUpdateEligibility) {
+          case SubscriptionUpdateEligibility.UPGRADE:
+            return 'iap_upgrade_contact_support';
+          case SubscriptionUpdateEligibility.BLOCKED_IAP:
+          default:
+            return 'iap_upgrade_already_subscribed';
+        }
```

Comment:
As mentioned in my other comment RE: typings - the `getSubscriptionUpdateEligibility` util does not return `SubscriptionUpdateEligibility.UPGRADE`/`SubscriptionUpdateEligibility.BLOCKED_IAP`